src/backend/foreign_key_builder.rs

```diff
@@ -30,19 +39,23 @@ pub trait ForeignKeyBuilder: QuotedBuilder {
         .unwrap()
     }
 
-    /// Translate [`ForeignKeyDropStatement`] into SQL statement.
-    fn prepare_foreign_key_drop_statement(
+    #[doc(hidden)]
+    /// Internal function to factor foreign key drop in table and outside.
+    fn prepare_foreign_key_drop_statement_internal(
         &self,
         drop: &ForeignKeyDropStatement,
         sql: &mut SqlWriter,
+        inside_table_single_alter: bool,
     );
 
+
     #[doc(hidden)]
     /// Internal function to factor foreign key creation in table and outside.
     fn prepare_foreign_key_create_statement_internal(
         &self,
         create: &ForeignKeyCreateStatement,
         sql: &mut SqlWriter,
         inside_table_creation: bool,
+        inside_table_single_alter: bool,
```

Comment:
It might be better to remove `inside_table_creation` and `inside_table_single_alter` and add argument mode with type: 
```
enum Mode {
    Creation,
    Alter,
    ...
}
```